packages/integration-tests/test/actions.spec.ts

```diff
@@ -65,27 +70,118 @@ const deposit = async (amount, value) => {
+    )
+  })
 
   it('should initialize the watcher', async () => {
     watcher = await initWatcher()
   })
 
   it('should deploy the simple storage contract', async () => {
-    SimpleStorage = await deploySimpleStorage()
+    simpleStorage = await simpleStorageFactory.deploy()
+    await simpleStorage.deployTransaction.wait()
   })
 
   it('should deposit from L1->L2', async () => {
     const value = `0x${'42'.repeat(32)}`
     await deposit(1, value)
-    const msgSender = await SimpleStorage.msgSender()
-    const l1ToL2Sender = await SimpleStorage.l1ToL2Sender()
-    const storageVal = await SimpleStorage.value()
-    const count = await SimpleStorage.totalCount()
+    const msgSender = await simpleStorage.msgSender()
+    const l1ToL2Sender = await simpleStorage.l1ToL2Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
 
     msgSender.should.be.eq(l2MessengerAddress)
     l1ToL2Sender.should.be.eq(l1Wallet.address)
     storageVal.should.be.eq(value)
     count.toNumber().should.be.eq(1)
   })
+
+  // TODO: Enable withdrawal test
+  it.skip('should withdraw from L2->L1', async () => {
+    const value = `0x${'77'.repeat(32)}`
+    await withdraw(value)
+    const msgSender = await simpleStorage.msgSender()
+    const l2ToL1Sender = await simpleStorage.l2ToL1Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
+
+    msgSender.should.be.eq(l1MessengerAddress)
+    l2ToL1Sender.should.be.eq(l2Wallet.address)
+    storageVal.should.be.eq(value)
+    count.toNumber().should.be.eq(1)
+  })
+})
+
+describe('ERC20', async () => {
+  const alice = new Wallet(etherbase, l2Provider)
+  const INITIAL_AMOUNT = 1000
+  const NAME = 'OVM Test'
+  const DECIMALS = 8
+  const SYMBOL = 'OVM'
+
+  it('should deploy the erc20 contract', async () => {
+    erc20 = await ERC20Factory.deploy(
+      INITIAL_AMOUNT, NAME, DECIMALS, SYMBOL
+    )
+  })
+
+  it('should set the total supply', async () => {
+    const totalSupply = await erc20.totalSupply()
+    expect(totalSupply.toNumber()).to.equal(INITIAL_AMOUNT)
+  })
+
+  it('should get the token name', async () => {
+    const name = await erc20.name()
+    expect(name).to.equal(NAME)
+  })
+
+  it('should get the token decimals', async () => {
+    const decimals = await erc20.decimals()
+    expect(decimals).to.equal(DECIMALS)
+  })
+
+  it('should get the token symbol', async () => {
+    const symbol = await erc20.symbol()
+    expect(symbol).to.equal(SYMBOL)
+  })
+
+  it('should assign initial balance', async () => {
+    const balance = await erc20.balanceOf(l2Wallet.address)
+    expect(balance.toNumber()).to.equal(INITIAL_AMOUNT)
+  })
+
+  it('should transfer amount to destination account', async () => {
+    const transfer = await erc20.transfer(alice.address, 100)
+    const receipt = await transfer.wait()
+
+    // There are two events from the transfer with the first being
```

Comment:
Good comment. Eventually we will want to remove the event for the fee because users will not expect it, so in the future that change will break this test

Reply:
We should assert there are only 2 events

Reply:
Good call. I'll add a check for that